app/uk/gov/hmrc/fileupload/ApplicationModule.scala

```diff
@@ -0,0 +1,151 @@
+
+  lazy val s3Service = new S3JavaSdkService(configuration.underlying, metrics.defaultRegistry)
+
+  lazy val downloadFromTransient = s3Service.downloadFromTransient
+
+  lazy val uploadToQuarantine = s3Service.uploadToQuarantine
+
+  lazy val deleteObjectFromQuarantineBucket: DeleteFileFromQuarantineBucket = s3Service.deleteObjectFromQuarantine
+
+  lazy val createS3Key = S3Key.forEnvSubdir(s3Service.awsConfig.envSubdir)
+
+  val redirectionFeature = new RedirectionFeature(configuration.underlying, httpErrorHandler)
+
+  lazy val zipAndPresign = s3Service.zipAndPresign _
+
+  val downloadFromQuarantine = s3Service.downloadFromQuarantine
+
+  lazy val loggerHelper: LoggerHelper = new LoggerHelperFileExtensionAndUserAgent
+
+  lazy val fileUploadBackendBaseUrl = servicesConfig.baseUrl("file-upload-backend")
+
+  lazy val testOnlySdesStubIsEnabled = servicesConfig.getConfBool("sdes-stub.enabled", defBool = false)
+
+  lazy val optTestOnlySdesStubBaseUrl = if (testOnlySdesStubIsEnabled) Some(servicesConfig.baseUrl("sdes-stub")) else None
+
+
+  val subscribe: (ActorRef, Class[_]) => Boolean = actorSystem.eventStream.subscribe
+
+  val publish: (AnyRef) => Unit = actorSystem.eventStream.publish
+
+  val now: () => Long = () => System.currentTimeMillis()
+
+  lazy val commandHandler: CommandHandler = new CommandHandlerImpl(auditedHttpExecute, fileUploadBackendBaseUrl, wsClient, publish)
+
+  lazy val getFileLength =
+    (envelopeId: EnvelopeId, fileId: FileId, version: FileRefId) =>
+      s3Service.getFileLengthFromQuarantine(createS3Key(envelopeId, fileId), version.value)
+
+  // scanner
+  actorSystem.actorOf(ScannerActor.props(subscribe, scanBinaryData, commandHandler), "scannerActor")
+  actorSystem.actorOf(TransferActor.props(subscribe, createS3Key, commandHandler, getFileLength, s3Service.copyFromQtoT), "transferActor")
+  actorSystem.actorOf(DeletionActor.props(subscribe, deleteObjectFromQuarantineBucket, createS3Key), "deletionActor")
+
+
+  //lazy val retrieveFileFromQuarantineBucket = s3Service.retrieveFileFromQuarantine _
+  //TODO discuss alternative thread pools
+  lazy val getFileFromQuarantine = {
+    lazy val ec = ExecutionContext.fromExecutorService(Executors.newFixedThreadPool(25))
+    QuarantineService.getFileFromQuarantine(s3Service.retrieveFileFromQuarantine)(_: S3KeyName, _: String)(ec)
+  }
+
+  // auditing
+  lazy val auditedHttpExecute = PlayHttp.execute(auditConnector, "file-upload-frontend", Some(t => logger.warn(t.getMessage, t))) _
+
+  lazy val auditF: (Boolean, Int, String) => (Request[_]) => Future[AuditResult] =
+    PlayHttp.audit(auditConnector, "file-upload-frontend", Some(t => logger.warn(t.getMessage, t)))
+
+  val auditedHttpBodyStreamer = (baseUrl: String, envelopeId: EnvelopeId, fileId: FileId, fileRefId: FileRefId, request: Request[_]) =>
+    new HttpStreamingBody(
+      url = s"$baseUrl/file-upload/envelopes/${envelopeId.value}/files/${fileId.value}/${fileRefId.value}",
```

Comment:
are these values all generated internally? wondering if we need to escape any url unsafe character that could break the path?